src/write.c

```diff
@@ -669,11 +671,11 @@ static avifResult avifEncoderAddImageInternal(avifEncoder * encoder,
         return AVIF_RESULT_NO_CODEC_AVAILABLE;
     }
 
-    avifEncoderChanges encoderChanges = 0;
-    if (!avifEncoderSettingsChanged(encoder, &encoderChanges)) {
+    avifEncoderChanges encoderChanges;
```

Comment:
Note that I removed the initialization of this variable, to test that the `avifEncoderDetectChanges()` call below sets its output parameter properly.